crates/biome_graphql_parser/src/parser/definitions/operation.rs

```diff
@@ -69,69 +62,213 @@ impl ParseRecovery for SelectionListParseRecovery {
+    const LIST_KIND: Self::Kind = GRAPHQL_VARIABLE_DEFINITION_LIST;
 
     fn parse_element(&mut self, p: &mut Self::Parser<'_>) -> ParsedSyntax {
-        parse_selection(p)
+        parse_variable_definition(p)
     }
 
     fn is_at_list_end(&self, p: &mut Self::Parser<'_>) -> bool {
-        p.at(T!['}'])
+        p.at(T![')'])
     }
 
     fn recover(
         &mut self,
         p: &mut Self::Parser<'_>,
         parsed_element: ParsedSyntax,
     ) -> biome_parser::parse_recovery::RecoveryResult {
-        parsed_element.or_recover(p, &SelectionListParseRecovery, expected_any_selection)
+        parsed_element.or_recover(
+            p,
+            &VariableDefinitionListParseRecovery,
+            expected_variable_definition,
+        )
     }
 }
 
+struct VariableDefinitionListParseRecovery;
+
+impl ParseRecovery for VariableDefinitionListParseRecovery {
+    type Kind = GraphqlSyntaxKind;
+    type Parser<'source> = GraphqlParser<'source>;
+    const RECOVERED_KIND: Self::Kind = GRAPHQL_VARIABLE_DEFINITION;
+
+    fn is_at_recovered(&self, p: &mut Self::Parser<'_>) -> bool {
+        is_at_variable(p)
+    }
+}
+
+/// https://spec.graphql.org/October2021/#sec-Language.Operations.Query-shorthand
+#[inline]
+pub(crate) fn parse_operation_definition(p: &mut GraphqlParser) -> ParsedSyntax {
+    if !is_at_operation(p) {
+        return Absent;
+    }
+
+    if is_at_selection_set(p) {
+        return parse_selection_set(p);
+    }
+
+    let m = p.start();
+    {
+        let m = p.start();
+        p.bump_ts(OPERATION_TYPE);
+        m.complete(p, GRAPHQL_OPERATION_TYPE);
+    }
+
+    // we don't need diagnostic here, because name is optional
+    parse_name(p).ok();
+    // we don't need diagnostic here, because variable definitions are optional
+    parse_variable_definitions(p).ok();
+
+    DirectiveList.parse_list(p);
+    parse_selection_set(p).ok();
+
+    Present(m.complete(p, GRAPHQL_OPERATION_DEFINITION))
+}
+
+#[inline]
+fn parse_selection_set(p: &mut GraphqlParser) -> ParsedSyntax {
+    let m = p.start();
+    p.expect(T!['{']);
+    SelectionList.parse_list(p);
+    p.expect(T!['}']);
+    Present(m.complete(p, GRAPHQL_SELECTION_SET))
+}
+
 #[inline]
 fn parse_selection(p: &mut GraphqlParser) -> ParsedSyntax {
-    // TODO: parse any selection
-    match p.cur() {
-        DOT3 => todo!(),
-        _ if is_at_field(p) => parse_field(p),
-        _ => Absent,
+    if is_at_field(p) {
+        parse_field(p)
+    } else if is_at_fragment(p) {
+        parse_fragment(p)
+    } else {
+        Absent
     }
 }
 
 #[inline]
 fn parse_field(p: &mut GraphqlParser) -> ParsedSyntax {
-    // TODO: parse alias, arguments, nested selection set
+    if !is_at_field(p) {
+        return Absent;
+    }
     let m = p.start();
-    parse_name(p).or_add_diagnostic(p, expected_name);
+    let next_token = p.lookahead();
+    if next_token == T![:] {
+        let m = p.start();
+        parse_name(p).ok();
+        p.bump(T![:]);
```

Comment:
I think you should use `.expect`, not bump because `:` is mandatory in the grammar

Reply:
`:` is mandatory in the gramma only if the field contain an alias, as `:` is a sign that signal the current name token is an alias for the next name token. Without the `:`, it's interpreted as two different field.
In this scenario, assume that `name1` is intended to be an alias for `name2`, but the `:` is missing:
```graphql
{
  name1 name2(abc: xyz)
}
```
It will be interpreted as this
```graphql
{
  name1,
  name2(abc: xyz)
}
```
I will make this behavior clearer in the code